### PR TITLE
165939888 User Should Be Able To View A Single Article Page

### PR DESCRIPTION
**Pull request by chikeozulumba**
#### Description
Articles form the backbone of our application and introduce us to CRUD which is the building block of most web applications.
#### Type of change
- [ ] Bug Fix (a non-breaking change which fixes an issue)
- [x] New Feature (a non-breaking change that adds functionality)
- [ ] New Chore (a non-breaking change that adds functionality)
- [ ] Breaking Change (a fix or feature which would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update
#### How Has This Been Tested?
- [ ] End to End
- [x] Integration
- [x] Unit test
#### Checklist:
- [ ] Unit test
- [ ] Code coverage
#### PT
[#165939888](https://www.pivotaltracker.com/story/show/165939888)
#### Screenshot
Screenshot 2019-05-19 at 12 29 55 AM

### CODE REVIEW

File: src/components/ArticleComponent/View/index.jsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React, { Fragment } from 'react';
+import Layout from './Layout';
+
+const ViewArticle = () => (
+  <Fragment>
```
Comment (ssanusi) [inline src/components/ArticleComponent/View/index.jsx:5]: @chikeozulumba why use fragment when only one component is to be rendered you might want to just use div

File: src/state/article/reducers.js [hunk 2]
```diff
@@ -1,28 +1,45 @@
...
+  IS_ARTICLE_REQUEST,
+  GET_ARTICLE_BY_ID_SUCCESS,
+  GET_ARTICLE_BY_ID_ERROR,
+} from './actionTypes';
+
 import { initialState } from './state';
 
 export default (state = initialState, action) => {
-  switch (action.type) {
+  if (action === undefined || !action) { return state; }
```
Comment (ssanusi) [inline src/state/article/reducers.js:12]: @chikeozulumba this line is not necessary if you have default in the switch statement

File: src/state/profile/actions.js [hunk 3]
```diff
@@ -10,10 +10,11 @@ export const getUserProfileFailure = error => ({
   payload: error,
 });
 
-export const getProfile = (route = 'profile') => async dispatch => {
+export const getProfile = (id = null, route = 'profile') => async dispatch => {
+  const profileURL = !id || id === undefined ? `/${route}` : `/${route}/${id}`;
```
Comment (ssanusi) [inline src/state/profile/actions.js:14]: @chikeozulumba am trying to understand this line of code why do we need to pass id and route as a parameter also I ready don't understand this line on code

File: src/state/article/reducers.js (hunk 2 above)
Comment (chikeozulumba) [reply src/state/article/reducers.js:12]: @ssanusi Noted

File: src/components/ArticleComponent/View/index.jsx (hunk 1 above)
Comment (chikeozulumba) [reply src/components/ArticleComponent/View/index.jsx:5]: @ssanusi Clarified

File: src/state/profile/actions.js (hunk 3 above)
Comment (chikeozulumba) [reply src/state/profile/actions.js:14]: @ssanusi Noted

**Review by micah-akpan — DISMISSED** [review]
LGTM

**Review by Oluwaseyi000 — COMMENTED** [review]
LGTM

**Comment by Raymond-Osy** [comment]
LGTM

**Review by Simplemart17 — COMMENTED** [review]
LGTM

**Review by Temmyogunbo — APPROVED** [review]

**Pull request merged**